velox/common/encode/Base64.h

```diff
@@ -89,19 +88,24 @@ class Base64 {
   static size_t
   decode(const char* src, size_t src_len, char* dst, size_t dst_len);
 
-  static void decodeUrl(
-      const char* src,
-      size_t src_len,
-      char* dst,
-      size_t dst_len,
-      bool pad);
+  static void
+  decodeUrl(const char* src, size_t src_len, char* dst, size_t dst_len);
 
   constexpr static char kBase64Pad = '=';
 
  private:
+  static inline bool isPadded(const char* data, size_t len) {
+    return (len > 0 && data[len - 1] == kPadding) ? true : false;
+  }
+
   static inline size_t countPadding(const char* src, size_t len) {
-    DCHECK_GE(len, 2);
-    return src[len - 1] != kBase64Pad ? 0 : src[len - 2] != kBase64Pad ? 1 : 2;
+    size_t padding_count = 0;
+    while (len > 0 && src[len - 1] == kPadding) {
```

Comment:
It was a mistake on my end, and I have corrected it. 

My actual task was to add the presto functions 
1. 'to_base32'(https://github.com/facebookincubator/velox/pull/8652)
2. 'from_base32'(https://github.com/facebookincubator/velox/pull/7672).

Since base64 and base32 share a lot of code, I was advised to create a utility class for the common code(https://github.com/facebookincubator/velox/pull/8650) and to clean base64(https://github.com/facebookincubator/velox/pull/8651)

While testing my functionality, I found the decoding in base64 had a bug! (https://github.com/facebookincubator/velox/issues/8646)

I had all these changes in one single PR, and @aditi-pandit suggested to separate out the PRs. while seperating out I missed to use kBase64Pad.

Reply:
@Joe-Abraham Got it. BTW, let's also update documentation for the affected function to clarify that it allows inputs without padding.